### PR TITLE
feat: cluster autoscaling

### PR DESCRIPTION
**Pull request by kolea2**

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java [hunk 1]
```diff
@@ -84,13 +84,60 @@ public CreateClusterRequest setZone(String zone) {
...
   public CreateClusterRequest setServeNodes(int numNodes) {
     proto.getClusterBuilder().setServeNodes(numNodes);
     return this;
   }
 
+  /**
+   * Sets the scaling node to manual and sets the number of nodes allocated to this cluster. More
+   * nodes enable higher throughput and more consistent performance.
+   */
+  public CreateClusterRequest setScalingMode(StaticClusterSize staticClusterSize) {
```
Comment (mutianf) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java:101]: Do we need `@Nonnull StaticClusterSize staticClusterSize` and `@Nonnull ClusterAutoscalingConfig autoscalingConfig` or check if `staticClusterSize` and `autoscalingConfig` is null first?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateInstanceRequest.java [hunk 2]
```diff
@@ -147,6 +147,34 @@ public CreateInstanceRequest addCluster(
...
+   * @param zone the zone where the cluster will be created.
+   * @param clusterAutoscalingConfig the autoscaling config that sets the min serve nodes, max serve
+   *     nodes, and CPU utilization percentage
+   * @param storageType the type of storage used by this cluster to serve its parent instance's
+   *     tables.
+   */
+  public CreateInstanceRequest addCluster(
+      @Nonnull String clusterId,
+      @Nonnull String zone,
+      ClusterAutoscalingConfig clusterAutoscalingConfig,
```
Comment (mutianf) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateInstanceRequest.java:166]: `@Nonnull` here also?

Comment (kolea2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateInstanceRequest.java]: good catch thanks!

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java (hunk 1 above)
Comment (kolea2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java:101]: thanks!

**Review by mutianf — APPROVED** [review]

**Pull request merged**